subsys/bluetooth/audio/ascs.c

```diff
@@ -122,6 +122,16 @@ void ascs_ep_set_state(struct bt_audio_ep *ep, uint8_t state)
 	       bt_audio_ep_state_str(old_state),
 	       bt_audio_ep_state_str(state));
 
+	if (ep->stream != NULL) {
+		if (state == BT_AUDIO_EP_STATE_CODEC_CONFIGURED &&
+		    old_state != BT_AUDIO_EP_STATE_IDLE) {
+			ascs_ep_unbind_audio_iso(ep);
+		} else if (state == BT_AUDIO_EP_STATE_RELEASING) {
+			ascs_ep_unbind_audio_iso(ep);
+			bt_audio_stream_detach(ep->stream);
+		}
+	}
```

Comment:
Can you elaborate on this change? 

Basically it does two things:
1) Add a `ep->stream` check which makes sense in some way, but really shouldn't be needed as an endpoint in a non-idle state should always have a stream, and if it doesn't it would indicate a problem elsewhere
2) It moves the check before the `if (ep->stream == NULL || old_state == state)` check, thus not handling this being called multiple times in the same state anymore if the state it is being moved to is `BT_AUDIO_EP_STATE_RELEASING`.

Reply:
You know, I looked into the code once again, and changed this a bit. I think that the change looks cleaner now.

> Can you elaborate on this change?

Basically I wanted to detach the stream before `ase_process` work is run. We hit race condition that `ase_process` is executed before `bt_audio_stream_detach(ep->stream);` thus below transition to idle is not executed:

```
	if (ase->ep.status.state == BT_AUDIO_EP_STATE_RELEASING &&
	    ase->ep.stream == NULL) {
 		ascs_ep_set_state(&ase->ep, BT_AUDIO_EP_STATE_IDLE);
 	}

```

Now, I've moved the bt_audio_stream_detach to be called from `ase_process` to eliminate the race